### PR TITLE
[MM-49009] Allow an end time for compliance exports

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
This is the server part of the new compliance export end time functionality. Compliance exports now allow the user to define a `--exportTo` flag with a Unix timestamp to limit the number of exports. The messages will be exported *excluding* the timestamp, so you can combine it with `--exportFrom` without an overlap.

It also logs the post IDs and timestamps for the first and last post in the export. Making it easier to do chunked exports.
```
{"timestamp":"2022-12-13 11:43:41.042 +01:00","level":"debug","msg":"Found posts to export","caller":"message_export/message_export.go:101","number_of_posts":199}
{"timestamp":"2022-12-13 11:43:41.042 +01:00","level":"info","msg":"Export scope","caller":"message_export/message_export.go:109","first_post_id":"oc7r4sdkm3gyzr98tg4qfphmqc","first_post_timestamp":1554596394957,"last_post_id":"aor98cf8mpyddckm4bo7cj8h6e","last_post_timestamp":1554597627720}
```

Despite what the CLI help text claimed, the timestamps are in unix milliseconds, not seconds. This has been corrected for the message exports.

#### Ticket Link
[MM-49009](https://mattermost.atlassian.net/browse/MM-49009)

#### Release Note
```release-note
Compliance exports may now be started with an end timestamp until which messages will be exported using the CLI: `--exportTo 123`
```

### CODE REVIEW

File: cmd/mattermost/commands/export.go [hunk 1]
```diff
@@ -168,6 +171,11 @@ func buildExportCmdF(format string) func(command *cobra.Command, args []string)
 			return errors.New("exportFrom must be a positive integer")
 		}
 
+		endTime, err := command.Flags().GetInt64("exportTo")
+		if err != nil {
+			return errors.New("exportFrom flag error")
+		}
+
 		limit, err := command.Flags().GetInt("limit")
```
Comment (isacikgoz) [inline cmd/mattermost/commands/export.go:179]: Can we return an error if both `limit` and `endTime` are greater than `0`? Or at least we can log a warning that results will be inconsistent.

**Review by isacikgoz — COMMENTED** [review]
Looks good but have a minor comment below and also can you add a specific test case for asserting the changes for adding `endAt` to the query?

File: cmd/mattermost/commands/export.go (hunk 1 above)
Comment (noxer) [reply cmd/mattermost/commands/export.go:179]: I'd rather just warn about this. There may be a use case where you want to limit to a timeframe or number of posts, whichever is lower.

Comment (isacikgoz) [reply cmd/mattermost/commands/export.go:179]: Yeah but what if there are multiple posts with the same timestamp? Then you'll need to use the postid as a cursor right?

Comment (noxer) [reply cmd/mattermost/commands/export.go:179]: Right. I'm not claiming to have a good use case for it but someone might. 2/5

**Comment by agnivade** [comment]
Let's discuss in the team channel.

**Review by agnivade — CHANGES_REQUESTED** [review]
Blocking this until we get some clarity on the problem.

**Pull request closed without merging**